controllers/devworkspace/sync/sync.go

```diff
@@ -91,15 +81,9 @@ func (s *Syncer) Delete(ctx context.Context, object metav1.Object) error {
 	return nil
 }
 
-func (s *Syncer) create(ctx context.Context, owner metav1.Object, key client.ObjectKey, blueprint metav1.Object) (runtime.Object, error) {
-	blueprintObject, ok := blueprint.(runtime.Object)
-	kind := blueprintObject.GetObjectKind().GroupVersionKind().Kind
-	if !ok {
-		return nil, fmt.Errorf("object %T is not a runtime.Object. Cannot sync it", blueprint)
-	}
-
-	actual := blueprintObject.DeepCopyObject()
-
+func (s *Syncer) create(ctx context.Context, owner client.Object, key client.ObjectKey, blueprint client.Object) (client.Object, error) {
+	actual := blueprint.DeepCopyObject().(client.Object)
+	kind := blueprint.GetObjectKind()
```

Comment:
I think "kind" we are using to log object, which will be created. "actual" variable that's copy of the original bluprint object. For multithreding code, I think that's a good idea to use copy of the object... But I think our controller code isn't async, so I guess we can use blueprint object directly. @metlos could you assist, please? Do you agree or maybe I missed something?

Reply:
I got it. Thank youy

Reply:
The copy is there so that we don't mutate the passed in `blueprint` object, but rather read into its copy, the `actual`. Not sure about the change from `metav1.Object` to `client.Object` but if it all compiles, then I guess we're fine.